### ISSUE TITLE
E0308 does not detected when returned value is inside `if ... {...}` without `else` 

### ISSUE THREAD

**Issue by neonaot** [intellij-rust/intellij-rust#8160]
## Environment

* **IntelliJ Rust plugin version:** 0.4.161.4264-213-nightly
* **Rust toolchain version:** 1.58.0-nightly (547a6ffee 2021-10-21) x86_64-apple-darwin
* **IDE name and version:** CLion 2021.3 RC (CL-213.5744.123)
* **Operating system:** macOS 11.4
* **Macro expansion engine:** new
* **Name resolution engine:** new

## Problem description

<img width="491" alt="изображение" src="https://user-images.githubusercontent.com/49211026/143208023-b45ff0b4-b544-4c57-b61e-3bb289eb122d.png">
I'm not sure if the title correctly describes this problem, feel free to change it.

Note, that this code Is correct and `f` will return `10` or `20`:
```rust
fn f(x: bool) -> u32 {
    if x {
        10
    } else { 
        20
    }
}
```

## Steps to reproduce
```rust
fn f(x: bool) -> i32{
     if x {
         -6
     }
     20
}
```
